app/daemon/daemon.go

```diff
@@ -112,105 +117,196 @@ func DownloadAndStart(serverConfig *conf.ServerConfig, userConfig *conf.Config)
 }
 
 
-func start(daemonPath string) (*exec.Cmd)  {
-	
-	log.Println("Booting deamon")
-	cmd := exec.Command(daemonPath)
-	cmd.Start()
-
-	return cmd
-
-}
-
 func Stop(cmd *exec.Cmd) {
 
 	if err := cmd.Process.Kill(); err != nil {
 		log.Fatal("failed to kill: ", err)
 	}
 }
 
-func NewDaemonAvailable(config *conf.ServerConfig) (bool, error) {
+
+func CheckForDaemon (serverConfig *conf.ServerConfig, userConfig *conf.Config) (string, error) {
 
 	log.Println("Checking daemon")
 
 	// get the latest release info
-	releaseVersion, versionErr := getReleaseVersion(config)
-	if versionErr != nil {
-		return false, versionErr
-	}
+	releaseVersion := userConfig.RunningNavVersion
 
-	log.Println("Latest version: " + releaseVersion)
+	log.Println("Looking for version: " + releaseVersion)
 
 	// get the apps current path
 	path, err := getCurrentPath()
 	if err != nil {
-		return false, err
+		return "", err
 	}
 
-	// append the deamon name
-
+	// build the path
 	path += "/lib/navcoin-" + releaseVersion + "/bin/"+ getOSInfo().DaemonName
 	log.Println("Looking for Navcoin Daemon at " + path )
 
-
 	// check the daemon exists
 	if !exists(path) {
 		log.Println("No Daemon found for version: " + releaseVersion)
-
-		return true, nil
+		return "", errors.New("No Daemon found for version: " + releaseVersion)
 	}else {
```

Comment:
There are a few places where you use `} else {` and in other places (like here) you use `}else {` 
Pretty anal but, it's nice to keep it consistent. Wonder if we could get a Go Linter running? (though from what I understand the compiler does some linting of it's own)

Reply:
fixed that instance, and i agree with you.

 You are right go has one build in but it doesn't catch things like that.  I will see what i can find for a linter.